lib/builder/builder.ex

```diff
@@ -0,0 +1,160 @@
+
+  There are 3 phases:
+
+  `:fetch` - This phase is responsible for downloading or copying in any replacement ERTS builds for cross-build targets.
+  `:patch` - The patch phase injects custom scripts into the build directory, this phase is also where any custom files should be copied into the build directory before being archived.
+  `:build` - This is the final phase in the build flow, it produces the final wrapper binary with a payload embedded inside.
+
+  You can add your own steps before and after phases execute. Your custom steps will also receive the build context struct, and can return a modified one to customize a build to your liking.
+
+  An example of adding a step before the fetch phase, and after the build phase:
+
+  ```
+  # ... mix.exs file
+  def releases do
+    [
+      my_app: [
+        steps: [:assemble, &Burrito.wrap/1],
+        burrito: [
+          # ... other Burrito configuration
+          extra_steps: [
+            fetch: [pre: [MyCustomStepModule, AnotherCustomStepModule]],
+            build: [post: [CustomStepAgain, YetAnotherCustomStepModule]]
+            # ...
+          ]
+        ]
+      ]
+    ]
+  end
+  # ...
+  ```
+  """
+
+  @phases [
+    fetch: [Fetch.InitBuild, Fetch.FetchERTS],
+    patch: [Patch.CopyERTS, Patch.CopyScripts, Patch.RecompileNIFs],
+    build: [Build.PackAndBuild, Build.CopyRelease]
+  ]
+
+  def build(%Mix.Release{} = release) do
+    options = release.options[:burrito] || []
+    debug? = Keyword.get(options, :debug, false)
+
+    build_targets = options[:targets]
+
+    # look for override target in system env
+    # if it's a valid target, set it as the only target
+    target_override_string = System.get_env("BURRITO_TARGET")
+
+    build_targets =
+      if target_override_string do
+        Log.warning(
+          :build,
+          "Target is being overridden with BURRITO_TARGET #{target_override_string}"
+        )
+
+        old_targets = Target.get_old_targets()
+
+        override_atom = try do
+          String.to_existing_atom(target_override_string)
+        rescue
+          _ -> exit_invalid_target(target_override_string)
+        end
+
+        # If we have a named target defined that matches this atom use that
+        # otherwise if it's a legacy build target, translate it (this will be removed soon!)
+        # otherwise :error, not a valid target
+        cond do
+          Keyword.has_key?(build_targets, override_atom) ->
+            Keyword.take(build_targets, [override_atom])
+
+          override_atom in old_targets ->
+            resolved_override = Target.maybe_translate_old_target(override_atom)
+            Keyword.put([], override_atom, resolved_override)
+
+          true ->
+            exit_invalid_target(override_atom)
+        end
+      else
+        build_targets
+      end
+
+    # Build every target
+    Enum.each(build_targets, fn {name, t} ->
```

Comment:
I think this will crash if neither a `targets` key or a target override are specified.

Should we default `build_targets` to `[]`, and print a warning right above this if `build_targets` is empty?